python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -271,7 +272,7 @@ def _get_table_slice(
         )
 
     def _check_supported_type(self, obj_type):
-        if obj_type not in self._handlers_by_type:
+        if not issubclass(obj_type, tuple(self._handlers_by_type.keys())):
```

Comment:
I have not measured any performance implications, but AFAIK this check is not hot so it should be negligible?